Lib/test/test_importlib/test_util.py

```diff
@@ -757,5 +757,49 @@ def test_source_from_cache_path_like_arg(self):
  ) = util.test_both(PEP3147Tests, util=importlib_util)
 
 
+class MagicNumberTests:
```

Comment:
I initially avoided this and followed the test machinery as I wasn't sure about directly importing `importlib.util` within this test module, but after looking at the way the test specialization works I don't see how this is an issue. Changed in 1e32a1b.